MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java

```diff
@@ -0,0 +1,1080 @@
+                    track.getAssignedForceReturnDates().get(forceID).isBefore(date)) {
+                forcesToUndeploy.add(forceID);
+            }
+        }
+        
+        for(int forceID : forcesToUndeploy) {
+            track.unassignForce(forceID);
+        }
+    }
+    
+    /**
+     * Processes an ignored Stratcon scenario
+     */
+    public static void processIgnoredScenario(StratconScenario scenario, StratconCampaignState campaignState) {
+        for(StratconTrackState track : campaignState.getTracks()) {
+            if(track.getScenarios().containsKey(scenario.getCoords())) {                
+                // subtract VP if scenario is 'required'
+                if(scenario.isRequiredScenario()) {
+                    campaignState.updateVictoryPoints(-1);
+                }
+                
+                StratconFacility localFacility = track.getFacility(scenario.getCoords());
+                if(localFacility != null) {
+                    // if the ignored scenario was on top of an allied facility
+                    // then it'll get captured, and the player will possibly lose a SO
+                    if(localFacility.getOwner() == ForceAlignment.Allied) {
+                        localFacility.setOwner(ForceAlignment.Opposing);
+                        
+                        if(localFacility.isStrategicObjective()) {
+                            campaignState.decrementStrategicObjectiveCompletedCount();
+                        }
+                    }
+                } else {
+                    // if it's an open-field
+                    // move scenario towards nearest allied facility
+                    StratconCoords closestAlliedFacilityCoords = track.findClosestAlliedFacilityCoords(scenario.getCoords());
+                    
+                    if(closestAlliedFacilityCoords != null) {
+                        StratconCoords newCoords = scenario.getCoords().translate(scenario.getCoords().direction(closestAlliedFacilityCoords));
+                        scenario.setCoords(newCoords);
+                        //scenario.setDeploymentDate(scenario.getDeploymentDate().plusDays(daysToAdd));
+                        
+                        // TODO: if the allied facility is in the new coords, replace this scenario
+                        // with a facility defense, with the opfor coming directly from all hostiles assigned to this scenario
+                        
+                        scenario.setCurrentState(ScenarioState.UNRESOLVED);
+                    } else {
+                        // TODO: if there's no allied facilities here, add its forces to track reinforcement pool
+                        
+                    }
+                }
+                
+                // either way, it's gone
+                track.removeScenario(scenario);
+            }
+        }
+    }
+    
+    public StratconRulesManager() {
+        MekHQ.registerHandler(this);
+    }
```

Comment:
I think this registration should be moved elsewhere rather than the constructor. Its an odd side effect to have here.